magenta/models/music_vae/data.py

```diff
@@ -1039,10 +1039,11 @@ def get_dataset(config, tf_file_reader_class=tf.data.TFRecordDataset,
 
   tf.logging.info('Reading examples from: %s', examples_path)
 
-  files = tf.data.Dataset.list_files(examples_path)
-  reader = files.apply(
-      tf.contrib.data.parallel_interleave(
-          tf_file_reader_class, cycle_length=num_threads))
+  filenames = tf.gfile.Glob(examples_path)
+  if is_training:
+    random.shuffle(filenames)
```

Comment:
would it be better to first do random.shuffle and then pass that to parallel_interleave?